### ISSUE TITLE
Possible bug: Using test set as validation for M4 (data leakage?)

### ISSUE THREAD

**Issue by Mango123L** [heyongxin233/DeTeCtive#7]
Hi, thanks for the work.

It seems the current M4 branch uses the test split as validation to select the best model.

Code (train_classifier.py):
```python
elif opt.dataset=='M4':
    dataset = load_M4(opt.path)
    passages_dataset = PassagesDataset(
        dataset[opt.database_name] + dataset[opt.database_name.replace('train','dev')],
        mode='M4'
    )
    val_dataset = PassagesDataset(dataset[opt.test_dataset_name], mode='M4')
```
Default script arguments set:
--database_name multilingual_train
--test_dataset_name multilingual_test

So:
- Training data = multilingual_train + multilingual_dev
- “Validation” (for best model) = multilingual_test (the test set)

**Question:**
Is this intentional, or should multilingual_dev be used for validation and multilingual_test only for final evaluation?

Please confirm whether this is a bug or design choice. Thanks!

**Comment by Mango123L** [comment]
I've noticed similar issues on other datasets. Is this a dataset leak, or am I misunderstanding something? Could you please explain? Thank you!

**Comment by heyongxin233** [comment]
@Mango123L Since the M4 training split contains relatively few generation models and its distribution differs substantially from the dev and test splits, the current design is intentional in order to obtain better performance under this setting. 

Regarding generalization, in our latest work [DETree](https://github.com/heyongxin233/DETree), we propose automatic data augmentation and more robust training and evaluation strategies, which lead to stronger cross-distribution generalization. please feel free to check it out.

**Comment by Mango123L** [comment]
Thank you for your response. I’ve also taken note of your DETree work, which is truly valuable.
